<?xml version="1.0" ?>
<launch>
  <arg name="config"		default="pgrp"/>
  <arg name="scene"		default="$(arg config)_calibration"/>
  <arg name="sim"		default="false"/>
  <arg name="robot_name"	default="a_bot"
       unless="$(eval arg('config') == 'ur5e')"/>
  <arg name="robot_name"	default="d_bot"
       if="$(eval arg('config') == 'ur5e')"/>
  <arg name="camera_name"	default="a_phoxi_m_camera"/>
  <arg name="world_frame"	default="workspace_center"/>
  <!-- <arg name="world_frame"	value="$(arg robot_name)_base_link"/> -->
  <arg name="camera_frame"	default="camera_link"/>
  <arg name="object_frame"	default="camera_marker"/>
  <arg name="reference_frame"	default=""/>

  <arg name="eye_on_hand"
       value="$(eval arg('camera_name') == arg('robot_name') + '_camera')"/>

  <!-- Start ArUco -->
  <include file="$(find o2as_easy_handeye)/launch/aruco_tracker.launch">
    <arg name="eye_on_hand"	value="$(arg eye_on_hand)"/>
    <arg name="camera_name"	value="$(arg camera_name)"/>
    <arg name="reference_frame"	value="$(arg reference_frame)"/>
    <arg name="camera_frame"	value="$(arg camera_frame)"/>
  </include>

  <!-- Start robot controller -->
  <include file="$(find o2as_easy_handeye)/launch/robot_controller.launch">
    <arg name="config"	value="$(arg config)"/>
    <arg name="scene"	value="$(arg scene)"/>
    <arg name="sim"	value="$(arg sim)"/>
    <arg name="rviz_config"
    	 value="$(find o2as_easy_handeye)/launch/o2as_easy_handeye.rviz"/>
  </include>

  <!-- Dummy calibration to have a fully connected TF tree and see all frames -->
  <node unless="$(arg eye_on_hand)"
	name="dummy_handeye" pkg="tf" type="static_transform_publisher"
        args="-0.425 0.45 0.88 -1.5707 0 -2.8 $(arg world_frame) $(arg camera_frame) 10" />
  <node if="$(arg eye_on_hand)"
	name="dummy_handeye" pkg="tf" type="static_transform_publisher"
        args="0 0 0.05 0 0 0 $(arg effector_frame) $(arg camera_frame) 10" />

  <!-- Set up namespace -->
  <arg name="namespace"	value="$(arg camera_name)_from_$(arg robot_name)"/>

  <group ns="$(arg namespace)" >
    <param name="eye_on_hand"		value="$(arg eye_on_hand)"/>
    <param name="world_frame"		value="$(arg world_frame)"/>
    <param name="effector_frame"	value="$(arg robot_name)_ee_link"
           unless="$(arg eye_on_hand)"/>
    <param name="robot_effector_frame"	value="$(arg robot_name)_wrist_2_link"
           if="$(arg eye_on_hand)"/>
    <param name="camera_frame"		value="$(arg camera_frame)" />
    <param name="object_frame"		value="$(arg object_frame)" />
  </group>

  <!-- Start easy_handeye calibrator -->
  <node name="handeye_calibrator" pkg="o2as_easy_handeye" type="calibrator"
  	ns="$(arg namespace)" output="screen"/>

  <!-- Start rqt_reconfigure -->
  <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure"/>
</launch>
